apps/extension/src/routes/popup/settings/settings-rpc.tsx

```diff
@@ -0,0 +1,86 @@
+import { AppQuerier } from 'penumbra-query/src/queriers/app';
+import { useState } from 'react';
+import { Button, FadeTransition, Input, InputProps } from 'ui';
+import { cn } from 'ui/lib/utils';
+import { useChainId } from '../../../hooks/chain-id';
+import { ShareGradientIcon } from '../../../icons';
+import { SettingsHeader } from '../../../shared';
+import { useStore } from '../../../state';
+import { networkSelector } from '../../../state/network';
+import { swClient } from '../../service-worker/extension/client/internal';
+
+export const SettingsRPC = () => {
+  const { chainId, refetch } = useChainId();
+  const { grpcEndpoint, setGRPCEndpoint } = useStore(networkSelector);
+
+  const [rpc, setRpc] = useState(grpcEndpoint ?? '');
+  const [rpcError, setRpcError] = useState(false);
+
+  const onChange: InputProps['onChange'] = e => {
```

Comment:
This runs everytime a keystroke happens. We should only run this after someone has clicked `save` button.